pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18886241?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18886241?buildTab=artifacts#/) on master @ [21e87bb80318f9dafdab356eb62c79b78eab12e2](https://github.com/cockroachdb/cockroach/commits/21e87bb80318f9dafdab356eb62c79b78eab12e2):


```
I250220 22:45:29.925967 18471 sql/catalog/lease/lease.go:480  [-] 2630  waiting for 1 leases to expire: desc=[{users 4 15}]
I250220 22:45:29.929550 29393 sql/catalog/lease/descriptor_state.go:135  [n1] 2631  new lease: tabledesc.immutable: {ID: 20, Version: 16, ModificationTime: "1740091529.894937749,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250220 22:45:29.930990 29429 sql/catalog/lease/descriptor_state.go:135  [n1] 2632  new lease: tabledesc.immutable: {ID: 23, Version: 16, ModificationTime: "1740091529.894937749,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:29.932465 29354 sql/catalog/lease/descriptor_state.go:135  [n1] 2633  new lease: tabledesc.immutable: {ID: 33, Version: 16, ModificationTime: "1740091529.894937749,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:29.933575 29208 sql/catalog/lease/descriptor_state.go:135  [n1] 2634  new lease: tabledesc.immutable: {ID: 44, Version: 16, ModificationTime: "1740091529.894937749,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250220 22:45:29.934766 27793 sql/catalog/lease/descriptor_state.go:135  [n1] 2635  new lease: tabledesc.immutable: {ID: 104, Version: 15, ModificationTime: "1740091529.894937749,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250220 22:45:29.935853 29446 sql/catalog/lease/descriptor_state.go:135  [n1] 2636  new lease: tabledesc.immutable: {ID: 107, Version: 15, ModificationTime: "1740091529.894937749,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250220 22:45:29.936887 29359 sql/catalog/lease/descriptor_state.go:135  [n1] 2637  new lease: tabledesc.immutable: {ID: 109, Version: 15, ModificationTime: "1740091529.894937749,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250220 22:45:29.945633 29554 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:56268,hostssl,user=reader] 2638 ={"Timestamp":1740091529945628702,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:56268","SessionID":"18260c06ee9a90880000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 22:45:29.951739 18667 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:56268,hostssl,user=reader] 2639 ={"Timestamp":1740091529951733045,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:56268","SessionID":"18260c06ee9a90880000000000000001","Duration":14011039}
I250220 22:45:30.173757 11946 kv/kvserver/queue.go:612  [T1,Vsystem,n1,s1,r80/1:/Tenant/10/Table/{12-25},raft] 2640  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I250220 22:45:30.470569 29803 sql/catalog/lease/descriptor_state.go:135  [n1] 2641  new lease: tabledesc.immutable: {ID: 4, Version: 17, ModificationTime: "1740091530.439247944,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250220 22:45:30.471687 18471 sql/catalog/lease/lease.go:480  [-] 2642  waiting for 1 leases to expire: desc=[{users 4 16}]
I250220 22:45:30.471792 29808 sql/catalog/lease/descriptor_state.go:135  [n1] 2643  new lease: tabledesc.immutable: {ID: 20, Version: 17, ModificationTime: "1740091530.439247944,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250220 22:45:30.473134 30303 sql/catalog/lease/descriptor_state.go:135  [n1] 2644  new lease: tabledesc.immutable: {ID: 23, Version: 17, ModificationTime: "1740091530.439247944,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:30.474737 30237 sql/catalog/lease/descriptor_state.go:135  [n1] 2645  new lease: tabledesc.immutable: {ID: 33, Version: 17, ModificationTime: "1740091530.439247944,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:30.476077 30341 sql/catalog/lease/descriptor_state.go:135  [n1] 2646  new lease: tabledesc.immutable: {ID: 44, Version: 17, ModificationTime: "1740091530.439247944,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250220 22:45:30.477357 30354 sql/catalog/lease/descriptor_state.go:135  [n1] 2647  new lease: tabledesc.immutable: {ID: 104, Version: 16, ModificationTime: "1740091530.439247944,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250220 22:45:30.478912 30359 sql/catalog/lease/descriptor_state.go:135  [n1] 2648  new lease: tabledesc.immutable: {ID: 107, Version: 16, ModificationTime: "1740091530.439247944,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250220 22:45:30.480090 30093 sql/catalog/lease/descriptor_state.go:135  [n1] 2649  new lease: tabledesc.immutable: {ID: 109, Version: 16, ModificationTime: "1740091530.439247944,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250220 22:45:30.493533 30673 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:56270,hostssl,user=reader] 2650 ={"Timestamp":1740091530493529478,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:56270","SessionID":"18260c070f46b5440000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 22:45:30.495650 30475 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:56270,hostssl,user=reader] 2651 ={"Timestamp":1740091530495645006,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:56270","SessionID":"18260c070f46b5440000000000000001","Duration":9770149}
I250220 22:45:30.979530 29834 sql/catalog/lease/descriptor_state.go:135  [n1] 2652  new lease: tabledesc.immutable: {ID: 4, Version: 18, ModificationTime: "1740091530.949296341,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250220 22:45:30.980562 18471 sql/catalog/lease/lease.go:480  [-] 2653  waiting for 1 leases to expire: desc=[{users 4 17}]
I250220 22:45:30.980731 29948 sql/catalog/lease/descriptor_state.go:135  [n1] 2654  new lease: tabledesc.immutable: {ID: 20, Version: 18, ModificationTime: "1740091530.949296341,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250220 22:45:30.981830 29953 sql/catalog/lease/descriptor_state.go:135  [n1] 2655  new lease: tabledesc.immutable: {ID: 23, Version: 18, ModificationTime: "1740091530.949296341,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:30.983163 30986 sql/catalog/lease/descriptor_state.go:135  [n1] 2656  new lease: tabledesc.immutable: {ID: 33, Version: 18, ModificationTime: "1740091530.949296341,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 22:45:30.984388 31111 sql/catalog/lease/descriptor_state.go:135  [n1] 2657  new lease: tabledesc.immutable: {ID: 44, Version: 18, ModificationTime: "1740091530.949296341,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250220 22:45:30.985488 31116 sql/catalog/lease/descriptor_state.go:135  [n1] 2658  new lease: tabledesc.immutable: {ID: 104, Version: 17, ModificationTime: "1740091530.949296341,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f573b0]

goroutine 30991 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x40077487a8?, {0x6892d78, 0x40087fbc70})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a20f80?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x40101f29a0?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6892d78, 0x40087fbc70})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6892d78?, 0x40087fbc70?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400fa395f0, {0x6892d78, 0x40087fbc70}, {0x400da42f18, 0x11}, 0x4007748f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x40083d8ed0, {0x6895db8?, 0x400cb680a0?}, 0x400614e2c0, {0x400adf8d78, 0x3}, {0x400fa395f0?, 0x54?}, 0x40066fdb60)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14983
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48010

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18886876?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18886876?buildTab=artifacts#/) on master @ [2e66bf7c1c1e67eaa11c2994380d467224863cb3](https://github.com/cockroachdb/cockroach/commits/2e66bf7c1c1e67eaa11c2994380d467224863cb3):


```
I250220 23:46:36.939298 37486 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58348,hostssl,user=reader] 3141 ={"Timestamp":1740095196939294433,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58348","SessionID":"18260f5cb8bf5d980000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:36.941259 37484 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58348,hostssl,user=reader] 3142 ={"Timestamp":1740095196941252800,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58348","SessionID":"18260f5cb8bf5d980000000000000001","Duration":5017777}
I250220 23:46:36.955358 37489 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58350,hostssl,user=reader] 3143 ={"Timestamp":1740095196955353240,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58350","SessionID":"18260f5cb9b57eda0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:36.957404 37487 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58350,hostssl,user=reader] 3144 ={"Timestamp":1740095196957399065,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58350","SessionID":"18260f5cb9b57eda0000000000000001","Duration":5033467}
I250220 23:46:36.971083 37619 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58352,hostssl,user=reader] 3145 ={"Timestamp":1740095196971079168,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58352","SessionID":"18260f5cbaa6f8510000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:36.973096 37604 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58352,hostssl,user=reader] 3146 ={"Timestamp":1740095196973090179,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58352","SessionID":"18260f5cbaa6f8510000000000000001","Duration":4899514}
I250220 23:46:36.988206 37607 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58354,hostssl,user=reader] 3147 ={"Timestamp":1740095196988201434,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58354","SessionID":"18260f5cbbab01530000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:36.990177 37605 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58354,hostssl,user=reader] 3148 ={"Timestamp":1740095196990172159,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58354","SessionID":"18260f5cbbab01530000000000000001","Duration":4940044}
I250220 23:46:37.004403 37609 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58356,hostssl,user=reader] 3149 ={"Timestamp":1740095197004398785,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58356","SessionID":"18260f5cbc9541870000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.006483 37637 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58356,hostssl,user=reader] 3150 ={"Timestamp":1740095197006478332,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58356","SessionID":"18260f5cbc9541870000000000000001","Duration":5894112}
I250220 23:46:37.029621 35653 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58358,hostssl,user=reader] 3151 ={"Timestamp":1740095197029616827,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58358","SessionID":"18260f5cbe21b2ac0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.033634 37638 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58358,hostssl,user=reader] 3152 ={"Timestamp":1740095197033628849,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58358","SessionID":"18260f5cbe21b2ac0000000000000001","Duration":7063492}
I250220 23:46:37.058558 36668 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58360,hostssl,user=reader] 3153 ={"Timestamp":1740095197058554216,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58360","SessionID":"18260f5cbfdbd4d50000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.060490 35917 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58360,hostssl,user=reader] 3154 ={"Timestamp":1740095197060484895,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58360","SessionID":"18260f5cbfdbd4d50000000000000001","Duration":4944004}
I250220 23:46:37.073718 36671 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58362,hostssl,user=reader] 3155 ={"Timestamp":1740095197073713874,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58362","SessionID":"18260f5cc0bdb44b0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.086098 37668 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58362,hostssl,user=reader] 3156 ={"Timestamp":1740095197086092944,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58362","SessionID":"18260f5cc0bdb44b0000000000000001","Duration":15748912}
I250220 23:46:37.099080 37694 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58364,hostssl,user=reader] 3157 ={"Timestamp":1740095197099076018,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58364","SessionID":"18260f5cc245a6910000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.103909 35920 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58364,hostssl,user=reader] 3158 ={"Timestamp":1740095197103903130,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58364","SessionID":"18260f5cc245a6910000000000000001","Duration":7872514}
I250220 23:46:37.118296 37696 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58366,hostssl,user=reader] 3159 ={"Timestamp":1740095197118290906,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58366","SessionID":"18260f5cc3699acc0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.120507 37717 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58366,hostssl,user=reader] 3160 ={"Timestamp":1740095197120501438,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58366","SessionID":"18260f5cc3699acc0000000000000001","Duration":5337319}
I250220 23:46:37.143548 37721 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58368,hostssl,user=reader] 3161 ={"Timestamp":1740095197143543465,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58368","SessionID":"18260f5cc4a8374c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250220 23:46:37.145455 35654 sql/catalog/lease/descriptor_state.go:135  [n1] 3162  new lease: tabledesc.immutable: {ID: 4, Version: 26, ModificationTime: "1740095197.107686088,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250220 23:46:37.146207 31259 sql/catalog/lease/lease.go:480  [-] 3163  waiting for 1 leases to expire: desc=[{users 4 25}]
I250220 23:46:37.146298 37642 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:58368,hostssl,user=reader] 3164 ={"Timestamp":1740095197146293734,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:58368","SessionID":"18260f5cc4a8374c0000000000000001","Duration":10249006}
I250220 23:46:37.147222 37746 sql/catalog/lease/descriptor_state.go:135  [n1] 3165  new lease: tabledesc.immutable: {ID: 20, Version: 26, ModificationTime: "1740095197.107686088,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250220 23:46:37.148686 37663 sql/catalog/lease/descriptor_state.go:135  [n1] 3166  new lease: tabledesc.immutable: {ID: 23, Version: 26, ModificationTime: "1740095197.107686088,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 23:46:37.151028 35761 sql/catalog/lease/descriptor_state.go:135  [n1] 3167  new lease: tabledesc.immutable: {ID: 33, Version: 26, ModificationTime: "1740095197.107686088,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250220 23:46:37.152867 37782 sql/catalog/lease/descriptor_state.go:135  [n1] 3168  new lease: tabledesc.immutable: {ID: 44, Version: 26, ModificationTime: "1740095197.107686088,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250220 23:46:37.155373 37787 sql/catalog/lease/descriptor_state.go:135  [n1] 3169  new lease: tabledesc.immutable: {ID: 104, Version: 25, ModificationTime: "1740095197.107686088,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f573b0]

goroutine 37810 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400dcc97a8?, {0x6892d78, 0x400dad9ea0})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a20f80?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x40018a3aa0?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6892d78, 0x400dad9ea0})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6892d78?, 0x400dad9ea0?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x4003de6630, {0x6892d78, 0x400dad9ea0}, {0x4002067038, 0x11}, 0x400dcc9f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400a0cee10, {0x6895db8?, 0x400a2cf040?}, 0x400aafa160, {0x4001ef60c0, 0x3}, {0x4003de6630?, 0x54?}, 0x40022367c8)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14492
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18887052?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18887052?buildTab=artifacts#/) on master @ [6bada5e7bc53457d56fc92acac8cac29127acbcb](https://github.com/cockroachdb/cockroach/commits/6bada5e7bc53457d56fc92acac8cac29127acbcb):


```
I250221 00:58:51.598686 18683 sql/catalog/lease/lease.go:480  [-] 2549  waiting for 1 leases to expire: desc=[{table_statistics 20 10}]
I250221 00:58:51.600346 24731 sql/catalog/lease/descriptor_state.go:135  [n1] 2550  new lease: tabledesc.immutable: {ID: 33, Version: 11, ModificationTime: "1740099531.538210884,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 00:58:51.601985 25464 sql/catalog/lease/descriptor_state.go:135  [n1] 2551  new lease: tabledesc.immutable: {ID: 44, Version: 11, ModificationTime: "1740099531.538210884,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 00:58:51.603806 25469 sql/catalog/lease/descriptor_state.go:135  [n1] 2552  new lease: tabledesc.immutable: {ID: 104, Version: 10, ModificationTime: "1740099531.538210884,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 00:58:51.606565 25490 sql/catalog/lease/descriptor_state.go:135  [n1] 2553  new lease: tabledesc.immutable: {ID: 107, Version: 10, ModificationTime: "1740099531.538210884,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 00:58:51.608264 25495 sql/catalog/lease/descriptor_state.go:135  [n1] 2554  new lease: tabledesc.immutable: {ID: 109, Version: 10, ModificationTime: "1740099531.538210884,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 00:58:51.619801 25358 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40054,hostssl,user=reader] 2555 ={"Timestamp":1740099531619796279,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40054","SessionID":"1826134df6d9d4810000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 00:58:51.627115 25312 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40054,hostssl,user=reader] 2556 ={"Timestamp":1740099531627109298,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40054","SessionID":"1826134df6d9d4810000000000000001","Duration":26950540}
I250221 00:58:52.798739 26138 sql/catalog/lease/descriptor_state.go:135  [n1] 2557  new lease: tabledesc.immutable: {ID: 4, Version: 12, ModificationTime: "1740099532.725710293,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 00:58:52.806488 26143 sql/catalog/lease/descriptor_state.go:135  [n1] 2558  new lease: tabledesc.immutable: {ID: 20, Version: 12, ModificationTime: "1740099532.725710293,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 00:58:52.811855 18683 sql/catalog/lease/lease.go:480  [-] 2559  waiting for 1 leases to expire: desc=[{table_statistics 20 11}]
I250221 00:58:52.822206 18683 sql/catalog/lease/lease.go:480  [-] 2560  waiting for 1 leases to expire: desc=[{role_members 23 11}]
I250221 00:58:52.817511 26062 sql/catalog/lease/descriptor_state.go:135  [n1] 2561  new lease: tabledesc.immutable: {ID: 23, Version: 12, ModificationTime: "1740099532.725710293,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 00:58:52.837973 24558 sql/catalog/lease/descriptor_state.go:135  [n1] 2562  new lease: tabledesc.immutable: {ID: 33, Version: 12, ModificationTime: "1740099532.725710293,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 00:58:52.839806 21181 sql/catalog/lease/descriptor_state.go:135  [n1] 2563  new lease: tabledesc.immutable: {ID: 44, Version: 12, ModificationTime: "1740099532.725710293,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 00:58:52.841637 26274 sql/catalog/lease/descriptor_state.go:135  [n1] 2564  new lease: tabledesc.immutable: {ID: 104, Version: 11, ModificationTime: "1740099532.725710293,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 00:58:52.843634 26291 sql/catalog/lease/descriptor_state.go:135  [n1] 2565  new lease: tabledesc.immutable: {ID: 107, Version: 11, ModificationTime: "1740099532.725710293,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 00:58:52.845482 26296 sql/catalog/lease/descriptor_state.go:135  [n1] 2566  new lease: tabledesc.immutable: {ID: 109, Version: 11, ModificationTime: "1740099532.725710293,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 00:58:52.850111 26259 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40056,hostssl,user=reader] 2567 ={"Timestamp":1740099532850107355,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40056","SessionID":"1826134e401fc7a30000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 00:58:52.868217 26234 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40056,hostssl,user=reader] 2568 ={"Timestamp":1740099532868211373,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40056","SessionID":"1826134e401fc7a30000000000000001","Duration":38731862}
I250221 00:58:54.002678 26808 sql/catalog/lease/descriptor_state.go:135  [n1] 2569  new lease: tabledesc.immutable: {ID: 4, Version: 13, ModificationTime: "1740099533.949680765,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 00:58:54.005013 26813 sql/catalog/lease/descriptor_state.go:135  [n1] 2570  new lease: tabledesc.immutable: {ID: 20, Version: 13, ModificationTime: "1740099533.949680765,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 00:58:54.007028 26998 sql/catalog/lease/descriptor_state.go:135  [n1] 2571  new lease: tabledesc.immutable: {ID: 23, Version: 13, ModificationTime: "1740099533.949680765,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 00:58:54.013165 18683 sql/catalog/lease/lease.go:480  [-] 2572  waiting for 1 leases to expire: desc=[{users 4 12}]
I250221 00:58:54.029001 27012 sql/catalog/lease/descriptor_state.go:135  [n1] 2573  new lease: tabledesc.immutable: {ID: 33, Version: 13, ModificationTime: "1740099533.949680765,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 00:58:54.041194 27026 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40058,hostssl,user=reader] 2574 ={"Timestamp":1740099534041189351,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40058","SessionID":"1826134e874a643b0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 00:58:54.045091 26816 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:40058,hostssl,user=reader] 2575 ={"Timestamp":1740099534045086715,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:40058","SessionID":"1826134e874a643b0000000000000001","Duration":21632372}
I250221 00:58:54.046694 27037 sql/catalog/lease/descriptor_state.go:135  [n1] 2576  new lease: tabledesc.immutable: {ID: 104, Version: 12, ModificationTime: "1740099533.949680765,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 00:58:54.049429 27058 sql/catalog/lease/descriptor_state.go:135  [n1] 2577  new lease: tabledesc.immutable: {ID: 107, Version: 12, ModificationTime: "1740099533.949680765,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f573b0]

goroutine 27063 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x40105987a8?, {0x6892db8, 0x400daee410})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a20f80?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x4007221960?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6892db8, 0x400daee410})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6892db8?, 0x400daee410?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x4006f7cb40, {0x6892db8, 0x400daee410}, {0x4009d0f008, 0x11}, 0x4010598f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x40079b9da0, {0x6895df8?, 0x400b29c320?}, 0x40037f0f20, {0x4001eb2aa8, 0x3}, {0x4006f7cb40?, 0x54?}, 0x400f5dc138)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14643
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18890163?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18890163?buildTab=artifacts#/) on master @ [b0b0e695f4623659574ebf45a5a140bb58673082](https://github.com/cockroachdb/cockroach/commits/b0b0e695f4623659574ebf45a5a140bb58673082):


```
I250221 11:59:48.651028 23527 sql/catalog/lease/descriptor_state.go:135  [n1] 2474  new lease: tabledesc.immutable: {ID: 23, Version: 10, ModificationTime: "1740139188.623104761,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:48.652318 22409 sql/catalog/lease/descriptor_state.go:135  [n1] 2475  new lease: tabledesc.immutable: {ID: 33, Version: 10, ModificationTime: "1740139188.623104761,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:48.653174 23183 sql/catalog/lease/descriptor_state.go:135  [n1] 2476  new lease: tabledesc.immutable: {ID: 44, Version: 10, ModificationTime: "1740139188.623104761,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 11:59:48.654171 23311 sql/catalog/lease/descriptor_state.go:135  [n1] 2477  new lease: tabledesc.immutable: {ID: 104, Version: 9, ModificationTime: "1740139188.623104761,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 11:59:48.655292 23588 sql/catalog/lease/descriptor_state.go:135  [n1] 2478  new lease: tabledesc.immutable: {ID: 107, Version: 9, ModificationTime: "1740139188.623104761,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 11:59:48.656353 23496 sql/catalog/lease/descriptor_state.go:135  [n1] 2479  new lease: tabledesc.immutable: {ID: 109, Version: 9, ModificationTime: "1740139188.623104761,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 11:59:48.670132 23792 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:34918,hostssl,user=reader] 2480 ={"Timestamp":1740139188670128297,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:34918","SessionID":"1826375f57f2bc580000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 11:59:48.672099 23790 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:34918,hostssl,user=reader] 2481 ={"Timestamp":1740139188672094081,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:34918","SessionID":"1826375f57f2bc580000000000000001","Duration":9869582}
I250221 11:59:49.076176 11081 2@util/log/event_log.go:39  [T1,Vsystem,n1] 2482 ={"Timestamp":1740139189076171480,"EventType":"runtime_stats","MemRSSBytes":433692672,"GoroutineCount":1246,"MemStackSysBytes":9601024,"GoAllocBytes":126558008,"GoTotalBytes":319324504,"HeapFragmentBytes":111231176,"HeapReservedBytes":52633600,"HeapReleasedBytes":52240384,"CGoAllocBytes":21622272,"CGoTotalBytes":54591488,"CGoCallRate":0.000000341352,"GCPausePercent":0.0000000009032252,"GCRunCount":29,"NetHostRecvBytes":61486458,"NetHostSendBytes":61457828}
I250221 11:59:49.076299 11081 2@server/status/runtime_log.go:43  [T1,Vsystem,n1] 2483  runtime stats: 414 MiB RSS, 1246 goroutines (stacks: 9.2 MiB), 121 MiB/304 MiB Go alloc/total (heap fragmentation: 106 MiB, heap reserved: 50 MiB, heap released: 50 MiB), 21 MiB/52 MiB CGO alloc/total (0.0 CGO/sec), 0.0/0.0 %(u/s)time, 0.0 %gc (29x), 59 MiB/59 MiB (r/w)net
I250221 11:59:49.101152 11549 3@pebble/event.go:1001  [n1,s1,pebble] 2484  [JOB 7] WAL created 000008
I250221 11:59:49.163290 24243 sql/catalog/lease/descriptor_state.go:135  [n1] 2485  new lease: tabledesc.immutable: {ID: 4, Version: 11, ModificationTime: "1740139189.137757072,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 11:59:49.164389 17866 sql/catalog/lease/lease.go:480  [-] 2486  waiting for 1 leases to expire: desc=[{users 4 10}]
I250221 11:59:49.164699 24299 sql/catalog/lease/descriptor_state.go:135  [n1] 2487  new lease: tabledesc.immutable: {ID: 20, Version: 11, ModificationTime: "1740139189.137757072,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 11:59:49.166190 24334 sql/catalog/lease/descriptor_state.go:135  [n1] 2488  new lease: tabledesc.immutable: {ID: 23, Version: 11, ModificationTime: "1740139189.137757072,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:49.167535 24370 sql/catalog/lease/descriptor_state.go:135  [n1] 2489  new lease: tabledesc.immutable: {ID: 33, Version: 11, ModificationTime: "1740139189.137757072,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:49.168905 24215 sql/catalog/lease/descriptor_state.go:135  [n1] 2490  new lease: tabledesc.immutable: {ID: 44, Version: 11, ModificationTime: "1740139189.137757072,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 11:59:49.170032 24362 sql/catalog/lease/descriptor_state.go:135  [n1] 2491  new lease: tabledesc.immutable: {ID: 104, Version: 10, ModificationTime: "1740139189.137757072,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 11:59:49.171303 24367 sql/catalog/lease/descriptor_state.go:135  [n1] 2492  new lease: tabledesc.immutable: {ID: 107, Version: 10, ModificationTime: "1740139189.137757072,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 11:59:49.172845 24375 sql/catalog/lease/descriptor_state.go:135  [n1] 2493  new lease: tabledesc.immutable: {ID: 109, Version: 10, ModificationTime: "1740139189.137757072,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 11:59:49.180712 24304 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:34920,hostssl,user=reader] 2494 ={"Timestamp":1740139189180708319,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:34920","SessionID":"1826375f76683ad60000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 11:59:49.184879 24302 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:34920,hostssl,user=reader] 2495 ={"Timestamp":1740139189184874173,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:34920","SessionID":"1826375f76683ad60000000000000001","Duration":11633233}
I250221 11:59:49.653621 24987 sql/catalog/lease/descriptor_state.go:135  [n1] 2496  new lease: tabledesc.immutable: {ID: 4, Version: 12, ModificationTime: "1740139189.627598216,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 11:59:49.654480 17866 sql/catalog/lease/lease.go:480  [-] 2497  waiting for 1 leases to expire: desc=[{users 4 11}]
I250221 11:59:49.655011 24992 sql/catalog/lease/descriptor_state.go:135  [n1] 2498  new lease: tabledesc.immutable: {ID: 20, Version: 12, ModificationTime: "1740139189.627598216,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 11:59:49.656211 23351 sql/catalog/lease/descriptor_state.go:135  [n1] 2499  new lease: tabledesc.immutable: {ID: 23, Version: 12, ModificationTime: "1740139189.627598216,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:49.657328 23357 sql/catalog/lease/descriptor_state.go:135  [n1] 2500  new lease: tabledesc.immutable: {ID: 33, Version: 12, ModificationTime: "1740139189.627598216,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 11:59:49.658312 23361 sql/catalog/lease/descriptor_state.go:135  [n1] 2501  new lease: tabledesc.immutable: {ID: 44, Version: 12, ModificationTime: "1740139189.627598216,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 11:59:49.659319 25142 sql/catalog/lease/descriptor_state.go:135  [n1] 2502  new lease: tabledesc.immutable: {ID: 104, Version: 11, ModificationTime: "1740139189.627598216,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f57450]

goroutine 25050 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4007a627a8?, {0x6893138, 0x400ea44a50})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a21260?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x4012217500?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6893138, 0x400ea44a50})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6893138?, 0x400ea44a50?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x40098386c0, {0x6893138, 0x400ea44a50}, {0x400ca7a318, 0x11}, 0x4007a62f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400a031e00, {0x6896178?, 0x400cf68aa0?}, 0x4009f014a0, {0x4008248cd8, 0x3}, {0x40098386c0?, 0x54?}, 0x400cdb57e8)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14415
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18891419?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18891419?buildTab=artifacts#/) on master @ [7bb86f65535c163a95bd6ec113dd3028e90db2ba](https://github.com/cockroachdb/cockroach/commits/7bb86f65535c163a95bd6ec113dd3028e90db2ba):


```
I250221 14:29:34.111802 41819 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45706,hostssl,user=reader] 3550 ={"Timestamp":1740148174111797824,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45706","SessionID":"18263f8b6e4be2f60000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.114333 41866 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45706,hostssl,user=reader] 3551 ={"Timestamp":1740148174114328157,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45706","SessionID":"18263f8b6e4be2f60000000000000001","Duration":5579074}
I250221 14:29:34.133441 41852 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45708,hostssl,user=reader] 3552 ={"Timestamp":1740148174133436347,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45708","SessionID":"18263f8b6f89deee0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.135963 41867 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45708,hostssl,user=reader] 3553 ={"Timestamp":1740148174135956982,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45708","SessionID":"18263f8b6f89deee0000000000000001","Duration":6368572}
I250221 14:29:34.150497 41823 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45710,hostssl,user=reader] 3554 ={"Timestamp":1740148174150492614,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45710","SessionID":"18263f8b709a159b0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.152541 41881 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45710,hostssl,user=reader] 3555 ={"Timestamp":1740148174152535895,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45710","SessionID":"18263f8b709a159b0000000000000001","Duration":5107447}
I250221 14:29:34.169090 41857 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45714,hostssl,user=reader] 3556 ={"Timestamp":1740148174169085483,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45714","SessionID":"18263f8b71b509b90000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.171317 41855 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45714,hostssl,user=reader] 3557 ={"Timestamp":1740148174171312140,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45714","SessionID":"18263f8b71b509b90000000000000001","Duration":5339759}
I250221 14:29:34.185251 41925 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45716,hostssl,user=reader] 3558 ={"Timestamp":1740148174185247012,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45716","SessionID":"18263f8b72ac3d7d0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.197185 41882 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45716,hostssl,user=reader] 3559 ={"Timestamp":1740148174197178927,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45716","SessionID":"18263f8b72ac3d7d0000000000000001","Duration":15006462}
I250221 14:29:34.221555 41760 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45720,hostssl,user=reader] 3560 ={"Timestamp":1740148174221550536,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45720","SessionID":"18263f8b74963d420000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.223905 41758 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45720,hostssl,user=reader] 3561 ={"Timestamp":1740148174223898997,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45720","SessionID":"18263f8b74963d420000000000000001","Duration":9613684}
I250221 14:29:34.246422 41986 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45722,hostssl,user=reader] 3562 ={"Timestamp":1740148174246418708,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45722","SessionID":"18263f8b76353c100000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.248662 41945 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45722,hostssl,user=reader] 3563 ={"Timestamp":1740148174248656482,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45722","SessionID":"18263f8b76353c100000000000000001","Duration":7174297}
I250221 14:29:34.266508 41929 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45724,hostssl,user=reader] 3564 ={"Timestamp":1740148174266503756,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45724","SessionID":"18263f8b7780176f0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.282330 41927 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45724,hostssl,user=reader] 3565 ={"Timestamp":1740148174282323434,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45724","SessionID":"18263f8b7780176f0000000000000001","Duration":19157998}
I250221 14:29:34.283105 42004 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45726,hostssl,user=reader] 3566 ={"Timestamp":1740148174283101250,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45726","SessionID":"18263f8b7876364a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.285338 41930 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45726,hostssl,user=reader] 3567 ={"Timestamp":1740148174285333005,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45726","SessionID":"18263f8b7876364a0000000000000001","Duration":6037673}
I250221 14:29:34.304584 12332 3@pebble/event.go:1001  [n1,s1,pebble] 3568  [JOB 8] WAL created 000009
I250221 14:29:34.323488 42029 sql/catalog/lease/descriptor_state.go:135  [n1] 3569  new lease: tabledesc.immutable: {ID: 4, Version: 30, ModificationTime: "1740148174.285546945,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 14:29:34.324585 31885 sql/catalog/lease/lease.go:480  [-] 3570  waiting for 1 leases to expire: desc=[{users 4 29}]
I250221 14:29:34.325018 42066 sql/catalog/lease/descriptor_state.go:135  [n1] 3571  new lease: tabledesc.immutable: {ID: 20, Version: 30, ModificationTime: "1740148174.285546945,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 14:29:34.325890 42036 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45730,hostssl,user=reader] 3572 ={"Timestamp":1740148174325886368,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45730","SessionID":"18263f8b7a0bfced0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:29:34.326573 42070 sql/catalog/lease/descriptor_state.go:135  [n1] 3573  new lease: tabledesc.immutable: {ID: 23, Version: 30, ModificationTime: "1740148174.285546945,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 14:29:34.328304 42075 sql/catalog/lease/descriptor_state.go:135  [n1] 3574  new lease: tabledesc.immutable: {ID: 33, Version: 30, ModificationTime: "1740148174.285546945,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 14:29:34.328405 42034 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45730,hostssl,user=reader] 3575 ={"Timestamp":1740148174328400002,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45730","SessionID":"18263f8b7a0bfced0000000000000001","Duration":22512024}
I250221 14:29:34.329581 42080 sql/catalog/lease/descriptor_state.go:135  [n1] 3576  new lease: tabledesc.immutable: {ID: 44, Version: 30, ModificationTime: "1740148174.285546945,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 14:29:34.330919 42045 sql/catalog/lease/descriptor_state.go:135  [n1] 3577  new lease: tabledesc.immutable: {ID: 104, Version: 29, ModificationTime: "1740148174.285546945,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 14:29:34.332878 41952 sql/catalog/lease/descriptor_state.go:135  [n1] 3578  new lease: tabledesc.immutable: {ID: 107, Version: 29, ModificationTime: "1740148174.285546945,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f589a0]

goroutine 42100 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x40081e47a8?, {0x68957b8, 0x4011da62d0})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a22fe0?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400eccba60?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x68957b8, 0x4011da62d0})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x68957b8?, 0x4011da62d0?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400fc13e60, {0x68957b8, 0x4011da62d0}, {0x400b95e7b0, 0x11}, 0x40081e4f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400ca4fa70, {0x68987f8?, 0x401099ba40?}, 0x401009e160, {0x4006daec60, 0x3}, {0x400fc13e60?, 0x54?}, 0x4008f557b8)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 15347
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18891634?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18891634?buildTab=artifacts#/) on master @ [fe860f65b9d31170610fd35fc3e334489c5bcf1b](https://github.com/cockroachdb/cockroach/commits/fe860f65b9d31170610fd35fc3e334489c5bcf1b):


```
I250221 14:26:12.002028 40866 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45734,hostssl,user=reader] 3323 ={"Timestamp":1740147972002024717,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45734","SessionID":"18263f5c5f9b72280000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.004457 40830 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45734,hostssl,user=reader] 3324 ={"Timestamp":1740147972004451893,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45734","SessionID":"18263f5c5f9b72280000000000000001","Duration":5609907}
I250221 14:26:12.019377 40750 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45736,hostssl,user=reader] 3325 ={"Timestamp":1740147972019372710,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45736","SessionID":"18263f5c60a554570000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.021804 40882 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45736,hostssl,user=reader] 3326 ={"Timestamp":1740147972021799007,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45736","SessionID":"18263f5c60a554570000000000000001","Duration":5532011}
I250221 14:26:12.035241 40753 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45738,hostssl,user=reader] 3327 ={"Timestamp":1740147972035237170,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45738","SessionID":"18263f5c6196b8250000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.037061 40751 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45738,hostssl,user=reader] 3328 ={"Timestamp":1740147972037055902,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45738","SessionID":"18263f5c6196b8250000000000000001","Duration":4969365}
I250221 14:26:12.051055 40884 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45740,hostssl,user=reader] 3329 ={"Timestamp":1740147972051051317,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45740","SessionID":"18263f5c628bf77c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.052949 40900 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45740,hostssl,user=reader] 3330 ={"Timestamp":1740147972052944386,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45740","SessionID":"18263f5c628bf77c0000000000000001","Duration":4785049}
I250221 14:26:12.066754 40888 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45742,hostssl,user=reader] 3331 ={"Timestamp":1740147972066749902,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45742","SessionID":"18263f5c63788ae20000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.069041 40886 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45742,hostssl,user=reader] 3332 ={"Timestamp":1740147972069036001,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45742","SessionID":"18263f5c63788ae20000000000000001","Duration":5373133}
I250221 14:26:12.090977 40932 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45744,hostssl,user=reader] 3333 ={"Timestamp":1740147972090973483,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45744","SessionID":"18263f5c64eb6ce80000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.093940 40959 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45744,hostssl,user=reader] 3334 ={"Timestamp":1740147972093935461,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45744","SessionID":"18263f5c64eb6ce80000000000000001","Duration":5966335}
I250221 14:26:12.107576 40935 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45746,hostssl,user=reader] 3335 ={"Timestamp":1740147972107571618,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45746","SessionID":"18263f5c65e9b1220000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.109756 40933 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45746,hostssl,user=reader] 3336 ={"Timestamp":1740147972109750961,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45746","SessionID":"18263f5c65e9b1220000000000000001","Duration":5117605}
I250221 14:26:12.126626 40905 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45748,hostssl,user=reader] 3337 ={"Timestamp":1740147972126621910,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45748","SessionID":"18263f5c670a976a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.131888 40961 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45748,hostssl,user=reader] 3338 ={"Timestamp":1740147972131883440,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45748","SessionID":"18263f5c670a976a0000000000000001","Duration":8316813}
I250221 14:26:12.148259 40879 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45750,hostssl,user=reader] 3339 ={"Timestamp":1740147972148255025,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45750","SessionID":"18263f5c6856ef880000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.150127 41010 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45750,hostssl,user=reader] 3340 ={"Timestamp":1740147972150121787,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45750","SessionID":"18263f5c6856ef880000000000000001","Duration":4774630}
I250221 14:26:12.164074 40881 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45752,hostssl,user=reader] 3341 ={"Timestamp":1740147972164070092,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45752","SessionID":"18263f5c694577520000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.166041 41011 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45752,hostssl,user=reader] 3342 ={"Timestamp":1740147972166035919,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45752","SessionID":"18263f5c694577520000000000000001","Duration":5056284}
I250221 14:26:12.188661 41029 sql/catalog/lease/descriptor_state.go:135  [n1] 3343  new lease: tabledesc.immutable: {ID: 4, Version: 29, ModificationTime: "1740147972.153544889,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 14:26:12.189473 31655 sql/catalog/lease/lease.go:480  [-] 3344  waiting for 1 leases to expire: desc=[{users 4 28}]
I250221 14:26:12.189949 40151 sql/catalog/lease/descriptor_state.go:135  [n1] 3345  new lease: tabledesc.immutable: {ID: 20, Version: 29, ModificationTime: "1740147972.153544889,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 14:26:12.191219 41048 sql/catalog/lease/descriptor_state.go:135  [n1] 3346  new lease: tabledesc.immutable: {ID: 23, Version: 29, ModificationTime: "1740147972.153544889,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 14:26:12.191745 40912 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45754,hostssl,user=reader] 3347 ={"Timestamp":1740147972191741907,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45754","SessionID":"18263f5c6ab311de0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 14:26:12.192779 41054 sql/catalog/lease/descriptor_state.go:135  [n1] 3348  new lease: tabledesc.immutable: {ID: 33, Version: 29, ModificationTime: "1740147972.153544889,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 14:26:12.194133 40656 sql/catalog/lease/descriptor_state.go:135  [n1] 3349  new lease: tabledesc.immutable: {ID: 44, Version: 29, ModificationTime: "1740147972.153544889,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 14:26:12.195197 41028 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:45754,hostssl,user=reader] 3350 ={"Timestamp":1740147972195192776,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:45754","SessionID":"18263f5c6ab311de0000000000000001","Duration":10253425}
I250221 14:26:12.195365 41016 sql/catalog/lease/descriptor_state.go:135  [n1] 3351  new lease: tabledesc.immutable: {ID: 104, Version: 28, ModificationTime: "1740147972.153544889,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f589e0]

goroutine 40142 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400e3af7a8?, {0x68958f8, 0x400601ab40})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a23120?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400e1ef240?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x68958f8, 0x400601ab40})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x68958f8?, 0x400601ab40?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400bfc7950, {0x68958f8, 0x400601ab40}, {0x40092d0198, 0x11}, 0x400e3aff40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400cf42480, {0x6898938?, 0x400e2646e0?}, 0x40092fcb00, {0x4001f1ee40, 0x3}, {0x400bfc7950?, 0x40?}, 0x400c018948)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 15055
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/lease/descriptor_set.go
+++ b/pkg/sql/catalog/lease/descriptor_set.go
@@ -108,10 +108,10 @@ func (l *descriptorSet) findPreviousToExpire(dropped bool) *descriptorVersionSta
 	}
 	exp.mu.Lock()
 	defer exp.mu.Unlock()
-	// If this version is not active, then it will go away on its own
-	// from the leases table, so no expiry needs to be setup. If the
-	// session is already cleared then an expiry has been setup too.
-	if exp.mu.refcount == 0 || exp.mu.session == nil {
+	// If the refcount has hit zero then this version will be cleaned up
+	// automatically. If an expiration time is already setup, then this
+	// version has already been expired.
+	if exp.mu.refcount == 0 || !exp.mu.expiration.IsEmpty() {
 		return nil
 	}
 	return exp

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -991,7 +991,6 @@ func purgeOldVersions(
 				if sessionExpiry := leaseToExpire.mu.session.Expiration(); leaseDuration > 0 && leaseToExpire.mu.expiration.Less(sessionExpiry) {
 					leaseToExpire.mu.expiration = sessionExpiry
 				}
-				leaseToExpire.mu.session = nil
 				if leaseToExpire.mu.lease != nil {
 					m.storage.sessionBasedLeasesWaitingToExpire.Inc(1)
 					m.mu.leasesToExpire = append(m.mu.leasesToExpire, leaseToExpire)

